### ISSUE TITLE
clarify docs about 'round()' accepting a negative integer for ndigits

### ISSUE THREAD

**Issue by b44b4b52-e24a-4005-a270-f6f76c384034** [python/cpython#73863]
BPO | [29677](https://bugs.python.org/issue29677)
--- | :---
Nosy | @mdickinson, @vstinner, @gerritholl, @vedgar, @Mariatta, @chris-rands, @csabella
PRs | <li>python/cpython#357</li><li>python/cpython#862</li><li>python/cpython#863</li><li>python/cpython#877</li><li>python/cpython#892</li><li>python/cpython#893</li>

<sup>*Note: these values reflect the state of the issue at the time it was migrated and might not reflect the current state.*</sup>

<details><summary>Show more details</summary><p>

GitHub fields:
```python
assignee = None
closed_at = <Date 2017-03-30.02:16:02.111>
created_at = <Date 2017-02-28.12:30:24.204>
labels = ['3.7', 'docs']
title = "clarify docs about 'round()' accepting a negative integer for ndigits"
updated_at = <Date 2017-03-30.02:16:02.110>
user = 'https://github.com/chris-rands'
```

bugs.python.org fields:
```python
activity = <Date 2017-03-30.02:16:02.110>
actor = 'Mariatta'
assignee = 'docs@python'
closed = True
closed_date = <Date 2017-03-30.02:16:02.111>
closer = 'Mariatta'
components = ['Documentation']
creation = <Date 2017-02-28.12:30:24.204>
creator = 'ChrisRands'
dependencies = []
files = []
hgrepos = []
issue_num = 29677
keywords = []
message_count = 18.0
messages = ['288710', '288711', '288713', '288714', '288715', '288751', '290644', '290662', '290669', '290670', '290671', '290672', '290694', '290793', '290794', '290802', '290803', '290804']
nosy_count = 8.0
nosy_names = ['mark.dickinson', 'vstinner', 'docs@python', 'Gerrit.Holl', 'veky', 'Mariatta', 'ChrisRands', 'cheryl.sabella']
pr_nums = ['357', '862', '863', '877', '892', '893']
priority = 'normal'
resolution = 'fixed'
stage = 'resolved'
status = 'closed'
superseder = None
type = None
url = 'https://bugs.python.org/issue29677'
versions = ['Python 3.5', 'Python 3.6', 'Python 3.7']
```

</p></details>

**Comment by b44b4b52-e24a-4005-a270-f6f76c384034** [comment]
With a negative integer for ndigits the output from 'round()', for example 'round(3, -2)', is always zero ('0' for 'int.\_\_round__' and '0.0' or '-0.0' for 'float.\_\_round__').

I think either it should raise an exception or the docs should be updated to reflect the current behavior.

The docs are currently silent on this: https://docs.python.org/3/library/functions.html#round

I don't know C, but there appears to be a note in the source about this implying it is the desired behavior but without an explanation: "For ndigits \< NDIGITS_MIN, x always rounds to +-0.0.": https://github.com/python/cpython/blob/6f0eb93183519024cb360162bdd81b9faec97ba6/Objects/floatobject.c

Anyway, I can't imagine an actual use case for negative ndigits.

**Comment by 6c53f22f-246d-4255-aab6-e377bdeb50f4** [comment]
```pycon
>>> round(21345, -2)
21300
```


Desired and useful to me.

**Comment by mdickinson** [comment]
Yep, it's (genuinely) a feature, not a bug, and being able to round to the nearest ten, hundred, thousand, etc. is useful. The docs could perhaps be improved to make it clearer that this is supported.

**Comment by b44b4b52-e24a-4005-a270-f6f76c384034** [comment]
Ah yes, you're both completely right of course. Perhaps the docs could still be clarified though. I managed to completely overlook this, and perhaps other non-expert Python users would too.

**Comment by mdickinson** [comment]
Yes, I think the initial description could be clarified. It currently reads:

> Return the floating point value number rounded to ndigits digits after the decimal point.

... which doesn't make it clear what happens for negative ndigits. 

There's a line in the second paragraph:

> values are rounded to the closest multiple of 10 to the power minus ndigits

which technically \*does* cover the case of negative ndigits correctly, but I think it would be good to have a clarification in the first paragraph of the description.

**Comment by fe5a23f9-4d47-49f8-9fb5-d6fbad5d9e38** [comment]
This is one of those things that show Python being extremely consistent within itself. Of course, docs should be clarified if needed, but the feature is desirable and useful.

**Comment by csabella** [comment]
I'm new to contributing.  Can I work on this one?

**Comment by vstinner** [comment]
New changeset 6003db7db5fec545c01923c198a5fdfca5a91538 by Victor Stinner (Gerrit Holl) in branch 'master':
bpo-29677: DOC: clarify documentation for `round` (#357)
https://github.com/python/cpython/commit/6003db7db5fec545c01923c198a5fdfca5a91538

**Comment by Mariatta** [comment]
New changeset 596506216104613591218c2896cdb49fa0b7e5bb by Mariatta in branch '3.6':
bpo-29677: DOC: clarify documentation for `round` (GH-357) (GH-862)
https://github.com/python/cpython/commit/596506216104613591218c2896cdb49fa0b7e5bb

**Comment by Mariatta** [comment]
New changeset 8994f36287b068e8ea9a9230fc90eda72bf5fff0 by Mariatta in branch '3.5':
bpo-29677: DOC: clarify documentation for `round` (GH-357) (GH-863)
https://github.com/python/cpython/commit/8994f36287b068e8ea9a9230fc90eda72bf5fff0

**Comment by Mariatta** [comment]
Cheryl, thank you for your interest in working on this issue.

The original issue has been addressed in Gerrit Holl's PR. But it seems like the docs can be further improved.

Here is Victor's comment about it: (copied from GH-357)

```
while we are modifying the doc, I dislike the start: "Return the floating point value number": number can be an integer. round(123, -2) doesn't use floating point numbers. Maybe rephrase to following text?

Return number rounded to (...). number can be an integer or a floating point number.

Or just remove "floating pointer number"?
```

Will you be interested in preparing a PR to address Victor's comment above?

Thanks :)

**Comment by Mariatta** [comment]
GH-357 is:
https://github.com/python/cpython/pull/357/files

:)

**Comment by csabella** [comment]
Mariatta,

Yes, thank you, I will work on a pull request to address Victor's comment.  Since it's my first time, I hope to read through the contributing docs and get this done today or tomorrow.

Thank you!

**Comment by Mariatta** [comment]
New changeset 85deefcf61d3cc192846f41a4ccc6df17da60c98 by Mariatta (csabella) in branch 'master':
bpo-29677: DOC: clarify documentation for `round` (GH-877)
https://github.com/python/cpython/commit/85deefcf61d3cc192846f41a4ccc6df17da60c98

**Comment by csabella** [comment]
Mariatta,

Thank you so much for your support and encouragement while I worked on my first PR.  You've made this an awesome experience!

**Comment by Mariatta** [comment]
New changeset 2609c9ee78c53d501914a5a90dbe094d9a8c3c97 by Mariatta in branch '3.6':
bpo-29677: DOC: clarify documentation for `round` (GH-877) (GH-892)
https://github.com/python/cpython/commit/2609c9ee78c53d501914a5a90dbe094d9a8c3c97

**Comment by Mariatta** [comment]
New changeset eef6e11f9883f54de491121b8c49fdadd3e3506d by Mariatta in branch '3.5':
bpo-29677: DOC: clarify documentation for `round` (GH-877) (GH-893)
https://github.com/python/cpython/commit/eef6e11f9883f54de491121b8c49fdadd3e3506d

**Comment by Mariatta** [comment]
Gerrit, Cheryl, thanks and congrats for your first contributions to CPython!

They've been merged and backported to 3.5 and 3.6.
So I'm closing this now.

Thanks all :)
